addons/delivery/models/delivery_carrier.py

```diff
@@ -123,3 +146,116 @@ def cancel_shipment(self, pickings):
         self.ensure_one()
         if hasattr(self, '%s_cancel_shipment' % self.delivery_type):
             return getattr(self, '%s_cancel_shipment' % self.delivery_type)(pickings)
+
+    @api.onchange('state_ids')
+    def onchange_states(self):
+        self.country_ids = [(6, 0, self.country_ids.ids + self.state_ids.mapped('country_id.id'))]
+
+    @api.onchange('country_ids')
+    def onchange_countries(self):
+        self.state_ids = [(6, 0, self.state_ids.filtered(lambda state: state.id in self.country_ids.mapped('state_ids').ids).ids)]
+
+    @api.multi
```

Comment:
hello @mge-odoo ,
IF I can use @api.one it's return the list of record set and i have to access this with list[0] and @api.multi is just return the record set so it's better to use api.multi and just check for single record using ensure_one(). so we need not to iterate the list .
Thanks